playbooks/replicate.yml

```diff
@@ -0,0 +1,86 @@
+- hosts: geniza_qa
+  connection: ssh
+  remote_user: pulsys
+  # override default location for db backup
+  vars:
+    - timestamp: "{{ ansible_date_time.date }}"
+    - db_backup_filename: "{{ timestamp }}_{{ application_db_name }}.sql.bz2"
+    - db_backup_path: "/home/{{ deploy_user }}/migrate/{{ db_backup_filename }}"
+    # testing replicating to localhost/dev for now
+    - dest_host: localhost
+    - dest_backup_path: "/tmp/migrate/{{ db_backup_filename }}"
+    - dest_backup_path_unzipped:  "{{ dest_backup_path | replace('.bz2', '') }}"
+    - dest_postgres_host: localhost
+  tasks:
+    - name: Generate a database dump
+      # use existing postgres backup task with customized backup path
+      # (role ensures backup dir exists)
+      include_role:
+        name: postgresql
+        tasks_from: backup_db.yml
+
+    - name: Create backup path on destination {{ dest_host }}
+      file:
+        dest: "{{ dest_backup_path | dirname }}"
+        mode: 0777
+        # owner: "{{ deploy_user }}"
+        state: directory
+      delegate_to: "{{ dest_host }}"
+
+    - name: Sync database dump to destination host
+      # run rsync on dest machine to pull db dump from the source machine
+      ansible.builtin.command: /usr/bin/rsync -avz "{{ ansible_user }}@{{ ansible_hostname }}:{{ db_backup_path }}" "{{ dest_backup_path }}" -e "ssh -o StrictHostKeyChecking=no"
+      # become: true
+      # become_user: "{{ deploy_user }}"
+      delegate_to: "{{ dest_host }}"
+
+    - name: Make sure unzipped database dump does not exist (if it does, bunzip2 will fail)
+      ansible.builtin.file:
+        path: "{{ dest_backup_path_unzipped }}"
+        state: absent
+      delegate_to: "{{ dest_host }}"
+
+    # postgres module can create zipped dump file, but at least on OSX
+    # it fails to unzip when attempting to restore;
+    # use command line since builtin unarchive task does not handle bz2
```

Comment:
Zip has the same problem (unarchive task does not handle it because it is not technically an archive like a tar file is). I went with bz2 because the compression is better and files are smaller than zip. Should I clarify in the comment?

Reply:
Ah, in that case I would probably just use gzip/tar. If you prefer bzip, a comment like "we use bzip for Reason X, using bzip means we have to use the command module to unzip" would preserve that thinking for Future Us.

Reply:
I messed with this a bit more and I think I remember more what the problem is — the postgres action can create a gzipped file (and also a tar file according to the docs); but the unarchive command doesn't seem able to handle the archive created by the postgres command. 

It does seem nicer to use archive/unarchive, though. Would it be better to just create a `.sql` file with the postgres action and then use `archive` and `unarchive?